### PR TITLE
[0.2] Creates tests + minor modifications

### PR DESCRIPTION
**Pull request by ghost**
The main purpose of this PR is to create unit tests for the library.

In addition:

1. I fixed some bugs
2. I improved the documentation
3. I improved the syntax according to PHP7
4. I made some refactoring

I put effort in respecting compatibility with the current version.

### CODE REVIEW

File: src/LinkedItems.php [hunk 1]
```diff
@@ -8,97 +8,122 @@
...
 
     /**
-     * Returns an array:
+     * Returns a sorted collection of linkable items and routes.
+     * This collection is used in the back office (backpack) to feed a select list.
+     * This select list is intended for adding new menu items.
+     *
+     * @return array
+     */
+    final public static function links(): array
```
Comment (jguyomard) [inline src/LinkedItems.php]: Why do you add this `final` keyword? I think we should allow the developper to override this method in his Model.

**Comment by jguyomard** [comment]
Thank you for this PR.
According to me, there are some breaking changes, we have to create a major release.

File: src/LinkedItems.php (hunk 1 above)
Comment (ghost) [reply src/LinkedItems.php]: I am not that sure! This method contains two static calls to methods which are both overridables. 
I think this gives the developer enough freedom.

Comment (jguyomard) [reply src/LinkedItems.php]: Personally, I hate these `final`. Let me override what i want! ;-)

Comment (tonyyb) [reply src/LinkedItems.php]: Agree with @jguyomard , `final` should be deleted.

**Review by tonyyb — COMMENTED** [review]
Good job ! 
After deleted `final` keyword seems OK for me !

**Comment by ghost** [comment]
Fair enough. I'll make the changes as soon as I can.

**Review by jguyomard — APPROVED** [review]

**Pull request merged**